src/effects/effectchainslot.cpp

```diff
@@ -158,21 +158,21 @@ void EffectChainSlot::slotChainChannelStatusChanged(const QString& group,
     }
 }
 
-void EffectChainSlot::slotChainEffectsChanged(bool shouldEmit) {
-    //qDebug() << debugString() << "slotChainEffectsChanged";
+void EffectChainSlot::slotChainEffectChanged(unsigned int effectSlotNumber,
+                                             bool shouldEmit) {
+    //qDebug() << debugString() << "slotChainEffectChanged" << effectSlotNumber;
     if (m_pEffectChain) {
         QList<EffectPointer> effects = m_pEffectChain->effects();
         if (effects.size() > m_slots.size()) {
             qWarning() << debugString() << "has too few slots for effect";
         }
-        for (int i = 0; i < m_slots.size(); ++i) {
-            EffectSlotPointer pSlot = m_slots[i];
-            EffectPointer pEffect;
-            if (i < effects.size()) {
-                pEffect = effects[i];
-            }
-            if (pSlot)
-                pSlot->loadEffect(pEffect);
+        EffectSlotPointer pSlot = m_slots[effectSlotNumber];
+        EffectPointer pEffect;
+        if (effectSlotNumber < (unsigned) effects.size()) {
+            pEffect = effects[effectSlotNumber];
```

Comment:
minor detail -- the QList above is now kind of unnecessary, but due to Qt copy-on-write semantics it's "free" -- but using ```operator[]``` counts as a write since it's a non-const method. If you use ```effects.at(effectSlotNumber)``` then it won't create a copy.

Alternately, you could add a ```EffectPointer EffectChain::effect(int)``` accessor method and drop the QList entirely.